### PR TITLE
Various testing improvements

### PR DESCRIPTION
**Pull request by brantburnett**
Motivation
----------
Improve ability to run tests.

Modifications
-------------
Upgrade to more recent testing frameworks.

Ensure that ClusterHelper is closed after integration tests so that the
process terminates cleanly.

Results
-------
Better testing during development.

### CODE REVIEW

File: Src/Couchbase.Linq.IntegrationTests/App.config [hunk 1]
```diff
@@ -39,7 +39,7 @@
       </dependentAssembly>
       <dependentAssembly>
         <assemblyIdentity name="Castle.Core" publicKeyToken="407dd0808d44fbdc" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-4.1.1.0" newVersion="4.1.1.0" />
+        <bindingRedirect oldVersion="0.0.0.0-4.2.1.0" newVersion="4.0.0.0" />
```
Comment (jeffrymorris) [inline Src/Couchbase.Linq.IntegrationTests/App.config]: What is the motivation for this change?

Comment (brantburnett) [reply Src/Couchbase.Linq.IntegrationTests/App.config]: @jeffrymorris 

As near as I can tell, they redid the DLL versioning between versions of Castle.Core and started versioning all of them as 4.0.0.0 for strong naming consistency.  Doesn't work without this binding redirect.

**Comment by jeffrymorris** [comment]
Thanks @brantburnett
